### ISSUE TITLE
P1 WARMUP: val_bpb=0.0000 printed for steps 1-8000 (artifact, NOT a leak)

### ISSUE THREAD

**Issue by gHashTag** [gHashTag/trios-trainer-igla#62]
## P1 — `val_bpb = 0.0000` printed for eval steps 1–8000 (warmup-print artifact)

Discovered while auditing `fix-verify-s43` (post-[#61](https://github.com/gHashTag/trios-trainer-igla/pull/61) byte-disjoint image).

### Observation (verbatim from `bpb_samples`)

```
canon_name         step   bpb
fix-verify-s43     1000   0.0003
fix-verify-s43     2000   0.0001
fix-verify-s43     3000   0.0000
fix-verify-s43     4000   0.0000
fix-verify-s43     5000   0.0000
fix-verify-s43     6000   0.0000
fix-verify-s43     7000   0.0000
fix-verify-s43     8000   0.0000
fix-verify-s43     9000   7.2781   <-- spike
fix-verify-s43    10000   1.6935
fix-verify-s43    11000   1.7399
fix-verify-s43    12000   1.5492   <-- honest, < 1.85
```

### Why this matters

On the pre-[#61](https://github.com/gHashTag/trios-trainer-igla/pull/61)
poisoned-corpus image, this artifact was indistinguishable from the
100 KB-prefix leak: both produce `val_bpb \u2248 0`. That made the original
leak diagnosis (`SCARABAEUS-LEAK-CONFIRMED-V2`) mass-flag **179 rows
too many**. After the corpus fix landed, `fix-verify-s43` proved the
artifact is trainer-side, not data-side.

Only 1 of 5 post-#61 runs reached step \u2265 9000 so far; the other 4 were
early-stopped inside the warmup-artifact zone and appear falsely as
leak. We need to suppress those prints to avoid contaminating future
ledgers.

### Likely root cause

In
[`src/train_loop.rs::run_single`](https://github.com/gHashTag/trios-trainer-igla/blob/main/src/train_loop.rs#L524)
the eval loop computes `val_bpb` by dividing per-token log-loss by the
number of evaluated tokens. In early steps the logits are still
near-zero so `softmax` returns \u2248 `1/VOCAB`; for the specific byte that
matches target the cross-entropy is `log(VOCAB) \u2248 5.5`, but the
printed value is aggregated differently \u2014 likely a **numerator/denominator
swap or a premature `break` before the accumulator fills**.

Concrete hypotheses (ranked):

1. **Eval returns early** if a sentinel token is hit within the first
   few windows, leaving `loss_sum/count = 0/count = 0`.
2. **`ema_bpb` initialised to `0.0` and masks `val_bpb`** \u2014 the printed
   `val_bpb` is actually the EMA, not the raw eval, for the first N
   evals.
3. **Warmup steps are skipped in eval** but `val_bpb` still prints the
   pre-init value.

### Fix proposal

- Investigate [`evaluate`](https://github.com/gHashTag/trios-trainer-igla/blob/main/src/train_loop.rs#L492)
  and assert `loss_sum > 0.0 && count > 0`; panic on the artifact.
- Add an integration test: run 500 steps with a fixed seed and assert
  the emitted `val_bpb` line is `> 0.5` within the first eval.
- Suppress `eprintln!("seed=... step=... val_bpb=...")` when
  `val_bpb < 1e-6` and log a warning instead.

### Acceptance

- [ ] Fresh `fix-verify-s44` (currently running) reports honest `val_bpb`
      at `step \u2265 1000` (no `0.0000` artifact zone)
- [ ] Integration test green on PR
- [ ] 179 Neon rows currently marked `SCARABAEUS-WARMUP-ARTIFACT`
      remain correctly labelled after fix (no mislabelling)

### Refs

- Parent: [trios-trainer-igla#61](https://github.com/gHashTag/trios-trainer-igla/pull/61) corpus fix
- Parent submission: [openai/parameter-golf#1998](https://github.com/openai/parameter-golf/pull/1998)
- Neon audit: `gardener_runs.action='gate2_first_honest_pass'` (2026-04-30 19:00 UTC)

phi^2 + phi^-2 = 3 \u00b7 R5-honest.
